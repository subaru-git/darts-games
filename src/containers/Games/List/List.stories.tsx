import type { Meta, StoryObj } from '@storybook/react';
import List from './List';

const meta: Meta<typeof List> = { component: List, title: 'Containers/Games/List' };

export default meta;
type Story = StoryObj<typeof List>;
export const Default: Story = {};
