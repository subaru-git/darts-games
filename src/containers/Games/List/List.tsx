import React, { FC } from 'react';
import { Flex, Heading, HStack, Text, Box } from '@chakra-ui/react';
import Link from 'next/link';
import AppTag from '@/components/AppTag';
import Card from '@/components/Card';
import useLocale from '@/hooks/locale';

const List: FC = () => {
  const { t } = useLocale();
  const data = Object.entries(t.games);
  return (
    <Box p={4}>
      <Heading as='h2' size={{ base: 'lg', md: 'xl' }} pb={4}>
        Games
      </Heading>
      <Flex flexWrap='wrap' gap={2} alignItems='stretch'>
        {data.map(([key, value]) => (
          <div key={key}>
            <Card>
              <Flex direction='column' gap={4}>
                <Link href={`/${key}`}>
                  <Heading as='h3' size={{ base: 'md', md: 'lg' }} p={{ base: 1, md: 4 }}>
                    {value.title}
                  </Heading>
                </Link>
                <HStack>
                  {value.tags.map((tag) => (
                    <AppTag key={tag} tag={tag} />
                  ))}
                </HStack>
                <Text whiteSpace='pre-wrap' fontSize={{ base: 'sm', md: 'md' }}>
                  {value.description.join('\n')}
                </Text>
              </Flex>
            </Card>
          </div>
        ))}
      </Flex>
    </Box>
  );
};

export default List;
