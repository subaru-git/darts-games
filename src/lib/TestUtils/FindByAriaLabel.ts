export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
