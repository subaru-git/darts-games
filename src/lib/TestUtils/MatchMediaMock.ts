import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
