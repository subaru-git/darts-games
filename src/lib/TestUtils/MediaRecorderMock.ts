export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
