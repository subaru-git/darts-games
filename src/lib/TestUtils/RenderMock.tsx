import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
