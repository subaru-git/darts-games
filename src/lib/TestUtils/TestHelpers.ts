import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}