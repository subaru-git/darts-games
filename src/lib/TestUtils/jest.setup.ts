import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
