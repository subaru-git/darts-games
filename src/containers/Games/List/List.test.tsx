import List from './List';
import { render } from '@/lib/TestUtils/RenderMock';

test('should rendering', () => {
  const { container } = render(<List />);
  expect(container).toMatchSnapshot();
});
