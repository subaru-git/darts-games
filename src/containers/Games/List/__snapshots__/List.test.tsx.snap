// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="css-h94677"
  >
    <h2
      class="chakra-heading css-uqd16f"
    >
      Games
    </h2>
    <div
      class="css-ojpwlm"
    >
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/eagleseye"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Eagle's Eye
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-blue-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                BULL
              </span>
              <span
                class="text-gray-100 bg-pink-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                count up
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              It is a count-up where only BULLs are added.
BULL is separate.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/cricketmarkup"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Cricket Mark-Up
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-teal-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                cricket
              </span>
              <span
                class="text-gray-100 bg-pink-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                count up
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              This game was introduced by "Kikuchi Yamaguchi."
First, Setting target count (recommended 10 times)
Shoot for each cricket numbers from 20 to 15 and Bull until the target count of times, like Cricket.
For more information, please watch this video.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/doubletrouble"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Double Trouble
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-cyan-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                double
              </span>
              <span
                class="text-gray-100 bg-pink-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                count up
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              You throw three darts at each double in turn, i.e. three darts at double one, three at double two etc through to double top.
Receive five points for every double you hit. 15 points available every 3darts; 300 overall.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/sweet16"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Sweet 16
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-cyan-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                double
              </span>
              <span
                class="text-gray-100 bg-orange-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                finish
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              You have 20 visits to the oche needing 32, which you have to finish in three darts with any combination finishing on a double.
Receive five points every time you finish 32, 15poits available every 3 darts; 300 overall.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/topsandtens"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Tops and Tens
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-cyan-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                double
              </span>
              <span
                class="text-gray-100 bg-orange-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                finish
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              You have 20 visits to the oche needing 40, which you have to finish in three darts or less only on double top, double ten or double five.
Receive five points every time you finish 40. 15 points available every 3darts, 300 overall.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/twodartcombinations"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Two-Dart Combinations
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-cyan-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                double
              </span>
              <span
                class="text-gray-100 bg-orange-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                finish
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              You have just three darts at each finish from 41-60, which you must finish on a double with any combination you wish.
Receive 15 points for every finish you complete. 15 points available every 3 darts; 300 overall.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/aroundthecompass"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Around The Compass
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-cyan-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                double
              </span>
              <span
                class="text-gray-100 bg-orange-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                finish
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              You have 20 visits to finish 24 using any combination but always finishing on a double, Recieve five points every time you complete the finish.
10point bonus for three double 12's (i.e. you get the double 12 first dart then follow it up with two more). 15 points available every 3 darts; 300 overall.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/tonsup"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Tons Up
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-cyan-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                double
              </span>
              <span
                class="text-gray-100 bg-orange-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                finish
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              You have 20 visits and your task is to finish 100 on a double using any combination as many times as possible.
Receive ten points every time you complete the finish and a ten-point bonus if you complete the finish in just two darts (i.e. treble 20 and double top) so 20 points available every 3 darts; 400 overall.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/route64"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Route 64
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-cyan-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                double
              </span>
              <span
                class="text-gray-100 bg-orange-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                finish
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              You have 20 visits to the oche needing 64, which you have to finish in three darts using any combination you wish.
Receive ten points for every finish you complete and a ten point bonus for first two dart finishes.
20 points available every 3 darts; 400 overall.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/eightythrew"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Eighty Threw
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-cyan-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                double
              </span>
              <span
                class="text-gray-100 bg-orange-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                finish
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              You have 20 visits to the oche needing 82, which is one of the most popular combinations and you have to finish in two or three darts using any method you wish. 
Receive ten points for every finish you complete and ten point bonuses for two darters.
20 points available every 3 darts; 400 overall.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/shanghainights"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Shanghai Nights
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-cyan-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                double
              </span>
              <span
                class="text-gray-100 bg-orange-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                finish
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              You have 20 visits and your task is to finish 120 as any times as possible.
Receive ten points every time you take out the combination. Ten points bonuses for three double-tops.
20 points available every 3 darts; 400 overall.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/switchhitter"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Switch Hitter
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-green-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                19
              </span>
              <span
                class="text-gray-100 bg-pink-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                count up
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              You have 60 darts at treble 19 - receive five points per treble 19, two points for a single 19 and a ten point bonus for a 171.
You receive no points for a dart which lands outside of the treble or single 19 beds.
25 points available every 3 darts; 500 overall.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/bullybully"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Bully Bully
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-blue-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                BULL
              </span>
              <span
                class="text-gray-100 bg-pink-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                count up
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              Throw 60 darts at the bullseye - earn five points for every bull hit and two for every outer bull.
10 point bonus for 3 bulls. 25 points available every 3 darts; 500 overall.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/treblesforshow"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Trebles For Show
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-green-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                20
              </span>
              <span
                class="text-gray-100 bg-pink-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                count up
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              Throw 60 darts at treble 20 - receive five points per treble 20. two points for a single 20 and a fifteen point bonus for a 180.
You receive no points for a dart which lands outside of the treble or single 20 beds. 30 points available every 3 darts; 600 overall.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/arrange"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Arrange
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-cyan-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                double
              </span>
              <span
                class="text-gray-100 bg-cyan-600 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                single
              </span>
              <span
                class="text-gray-100 bg-cyan-800 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                master
              </span>
              <span
                class="text-gray-100 bg-orange-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                finish
              </span>
              <span
                class="text-gray-100 bg-purple-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                simulation
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              Out for random scores with set options.
The simulation mode must include the specified range of deviations.
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="h-full max-w-md rounded-xl border border-gray-200 bg-gray-50 p-5 shadow-md"
        >
          <div
            class="css-1811skr"
          >
            <a
              href="/countup"
            >
              <h3
                class="chakra-heading css-dnq814"
              >
                Count Up
              </h3>
            </a>
            <div
              class="chakra-stack css-1igwmid"
            >
              <span
                class="text-gray-100 bg-pink-400 whitespace-nowrap rounded-full px-2.5 py-0.5 text-sm font-medium"
              >
                count up
              </span>
            </div>
            <p
              class="chakra-text css-12mgru9"
            >
              We will compete for the total score in 8 rounds.
In this game, since it is designed specifically for steel darts, I will only record the total value of each round.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
